### ISSUE TITLE
Lot of accessibility issues

### ISSUE THREAD

**Issue by mellis481** [react-component/pagination#429]
Certain implementation decisions that were made for the pagination component have resulted in some serious inaccessibility issues.  These include:

- The programmatic label for all pager buttons uses the `title` attribute which is sub-optimal. `aria-label` is the best attribute to use as it works with more screen readers.
- The programmatic label for page number pager buttons only includes the page number itself and cannot be changed so screen readers will only ever announce eg. "2" instead of something more helpful like "Page 2".
- Since a pagination component includes navigation elements, it should necessarily be wrapped in a `nav` element with appropriate programmatic label (eg. `<nav aria-label=”pagination”>`).
- The active page is not identified (using `aria-current="page"`) and is, therefore, not announced by screen readers.
- Instead of having the focusable/actionable pager elements be the anchor or button _within_ each `<li>` element, the focusable elements in the Pagination component _are_ the `<li>` elements. This results in the following issues:
  - a suboptimal screen reader UX because, when an element is focused, a screen reader will announce “clickable item 2” instead of something more helpful like "page 2 link".
  - the focusable element is only actionable via the ENTER key only rather than the SPACE and ENTER keys.
  - using UP/DOWN keys has unexpected/undesirable results.

For reference, [here](https://design-system.w3.org/components/pagination.html) is a good example of an accessible pagination component.

**Comment by SalahAdDin** [comment]
> Certain implementation decisions that were made for the pagination component have resulted in some serious inaccessibility issues. These include:
> 
>     * The programmatic label for all pager buttons uses the `title` attribute which is sub-optimal. `aria-label` is the best attribute to use as it works with more screen readers.
> 
>     * The programmatic label for page number pager buttons only includes the page number itself and cannot be changed so screen readers will only ever announce eg. "2" instead of something more helpful like "Page 2".
> 
>     * Since a pagination component includes navigation elements, it should necessarily be wrapped in a `nav` element with appropriate programmatic label (eg. `<nav aria-label=”pagination”>`).
> 
>     * The active page is not identified (using `aria-current="page"`) and is, therefore, not announced by screen readers.
> 
>     * Instead of having the focusable/actionable pager elements be the anchor or button _within_ each `<li>` element, the focusable elements in the Pagination component _are_ the `<li>` elements. This results in the following issues:
>       
>       * a suboptimal screen reader UX because, when an element is focused, a screen reader will announce “clickable item 2” instead of something more helpful like "page 2 link".
>       * the focusable element is only actionable via the ENTER key only rather than the SPACE and ENTER keys.
>       * using UP/DOWN keys has unexpected/undesirable results.
> 
> 
> For reference, [here](https://design-system.w3.org/components/pagination.html) is a good example of an accessible pagination component.

Is this library even mantianed nowadays?
